cpu/stm32_common/periph/rtc.c

```diff
@@ -322,4 +329,257 @@ void ISR_NAME(void)
+    {
+        DEBUG("[RTC initialize]\n");
+
+        // Resets the entire Backup domain
+        RCC->BDCR |= RCC_BDCR_BDRST;
+        // Reset not activated
+        RCC->BDCR &= ~RCC_BDCR_BDRST;
+
+        //RTC clock enabled
+        //LSE oscillator clock used as RTC clock10: LSI oscillator clock used as RTC clock
+        RCC->BDCR |= RCC_BDCR_RTCEN | RCC_BDCR_RTCSEL_LSE;
+
+        // Second interrupt is enabled.
+        RTC->CRH |= RTC_CRH_SECIE;
+
+        // Enter configuration mode.
+        RTC->CRL |= RTC_CRL_CNF;
+
+        //If the input clock frequency (fRTCCLK) is 32.768 kHz, write 7FFFh in this register to get a signal period of 1 second.
+        RTC->PRLH = 0;
+        RTC->PRLL = 0x8000;         //тактирование от внешнего кварца
+
+        // Calibration clock output disable (for enable in pin PC13 "AFIO->EVCR = 0xAD" and "BKP->RTCCR | = BKP_RTCCR_CCO | BKP_RTCCR_ASOS | BKP_RTCCR_ASOE;")
+        AFIO->EVCR = 0;
+        BKP->RTCCR = 0;
+
+        // Exit configuration mode (start update of RTC registers).
+        RTC->CRL &= ~RTC_CRL_CNF;
+
+        RCC->BDCR |= RCC_BDCR_LSEON;
+        while ((RCC->BDCR & RCC_BDCR_LSEON) != RCC_BDCR_LSEON){}
+
+        // Wait registers synchronize flag
+        RTC->CRL &= (uint16_t)~RTC_CRL_RSF;
+        while((RTC->CRL & RTC_CRL_RSF) != RTC_CRL_RSF){}
+    }
+}
+
+static void _rtc_enter_config_mode(void)
+{
+    // Wait until the RTOFF bit is 1 (no RTC register writes ongoing).
+    while ((RTC->CRL & RTC_CRL_RTOFF) == 0);
+
+    // Enter configuration mode.
+    RTC->CRL |= RTC_CRL_CNF;
+}
+
+static void _rtc_exit_config_mode(void)
+{
+    // Exit configuration mode.
+    RTC->CRL &= ~RTC_CRL_CNF;
+
+    // Wait until the RTOFF bit is 1 (our RTC register write finished).
+    while ((RTC->CRL & RTC_CRL_RTOFF) == 0);
+}
+
+static void _rtc_set_alarm_time(uint32_t alarm_time)
+{
+    _rtc_enter_config_mode();
+    RTC->ALRL = (alarm_time & 0x0000ffff);
+    RTC->ALRH = (alarm_time & 0xffff0000) >> 16;
+    _rtc_exit_config_mode();
+}
+
+static void _rtc_enable_alarm(void)
+{
+    _rtc_enter_config_mode();
+    RTC->CRH |= RTC_CRH_ALRIE;
+    _rtc_exit_config_mode();
+}
+
+static void _rtc_disable_alarm(void)
+{
+    _rtc_enter_config_mode();
+    RTC->CRH &= ~RTC_CRH_ALRIE;
+    _rtc_exit_config_mode();
+}
+
+static uint32_t rtc_get_counter_val(void)
+{
+    return (RTC->CNTH << 16) | RTC->CNTL;
+}
+
+static uint32_t rtc_get_alarm_val(void)
+{
+    return (RTC->ALRH << 16) | RTC->ALRL;
+}
+
+static void _rtc_set_counter_val(uint32_t counter_val)
+{
+    _rtc_enter_config_mode();
+    RTC->CNTH = (counter_val & 0xffff0000) >> 16; // CNT[31:16]
+    RTC->CNTL = counter_val & 0x0000ffff;         // CNT[15:0]
+    _rtc_exit_config_mode();
+}
+
+// (UnixTime = 00:00:00 01.01.1970 = JD0 = 2440588)
+#define JULIAN_DATE_BASE    2440588
+
+// Julian day number calculation - https://en.wikipedia.org/wiki/Julian_day
```

Comment:
Why should the RTC use the Julian calendar?

> The Gregorian calendar is the most widely used civil calendar in the world.

https://en.wikipedia.org/wiki/Gregorian_calendar